File: app/src/main/java/com/imangazalievm/bubbble/di/ShotsSearchPresenterComponent.java
package com.imangazalievm.bubbble.di;

import com.imangazalievm.bubbble.di.modules.ShotsSearchPresenterModule;
import com.imangazalievm.bubbble.di.scopes.Presenter;
import com.imangazalievm.bubbble.presentation.mvp.presenters.ShotsSearchPresenter;

import dagger.Component;

@Presenter
@Component(dependencies = ApplicationComponent.class, modules = ShotsSearchPresenterModule.class)
public interface ShotsSearchPresenterComponent {

    ShotsSearchPresenter getPresenter();

}

File: app/src/main/java/com/imangazalievm/bubbble/di/shotssearch/ShotsSearchPresenterComponent.kt
package com.imangazalievm.bubbble.di.shotssearch

import com.imangazalievm.bubbble.di.global.scopes.Presenter
import com.imangazalievm.bubbble.di.global.ApplicationComponent
import com.imangazalievm.bubbble.presentation.shotssearch.ShotsSearchPresenter
import dagger.Component

@Presenter
@Component(
    dependencies = [ApplicationComponent::class],
    modules = [ShotsSearchPresenterModule::class]
)
interface ShotsSearchPresenterComponent {

    fun getPresenter(): ShotsSearchPresenter

}